### PR TITLE
Throw error instead of emit error for fixing unexpected type

### PR DESCRIPTION
**Pull request by yuta0801**
### Description of the PR

Store#set() is expected to return `?Piece`, but may return boolean because of returns client.emit() directly.
EventEmitter.emit() returns true if the event had listeners, false otherwise.
See: https://nodejs.org/api/events.html#events_emitter_emit_eventname_args

Code affected by this behavior: https://github.com/dirigeants/klasa/blob/ea059931fad07bdea870620181de453efe816646/src/lib/structures/base/AliasStore.js#L50-L52

### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)

- Fixed unexpected return type of Store#set() becouse of returning the client.emit() directly

### Semver Classification

- [ ] This PR only includes documentation or non-code changes.
- [x] This PR fixes a bug and does not change the (intended) framework interface.
- [ ] This PR adds methods or properties to the framework interface.
- [ ] This PR removes or renames methods or properties in the framework interface.

### CODE REVIEW

**Comment by bdistin** [comment]
actually it should just throw a full error.

**Comment by yuta0801** [comment]
I think it rarely happens unexpected behavior, but it returns true when someone handling `error` event of client (`.emit()` only throws an error when no `error` event handler)
And at this time, pass the wrong Peace to `AliasStore#set()`, TypeError will occur because of `(true).aliases` is executed

**Comment by PyroTechniac** [comment]
That will cause more confusion and be harder for both users and framework developers to find the source of the error.

This is definitely a bug in the framework but I agree that it should just throw an error, as the client will stop loading if an error occurs (which it should)

**Comment by yuta0801** [comment]
Should I change to throw an error instead of explicitly returning null after emitting the error?

**Comment by bdistin** [comment]
yeah

File: src/lib/structures/base/Store.js [hunk 1]
```diff
@@ -133,13 +133,10 @@ class Store extends Collection {
...
 	 * @param {Piece} piece The piece we are setting up
-	 * @returns {?Piece}
+	 * @returns {Piece}
 	 */
 	set(piece) {
-		if (!(piece instanceof this.holds)) {
-			this.client.emit('error', `Only ${this} may be stored in this Store.`);
-			return null;
-		}
+		if (!(piece instanceof this.holds)) throw new Error(`Only ${this} may be stored in this Store.`);
```
Comment (yuta0801) [inline src/lib/structures/base/Store.js]: Is TypeError better than just Error?

Comment (PyroTechniac) [reply src/lib/structures/base/Store.js]: TypeError would be a more accurate error to use, yes

**Pull request merged**